src/elements/components/FeatheryAuthGate.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Form } from '../..';
+import { Props as FormProps } from '../../Form';
+import { getStytchJwt } from '../../utils/browser';
+import { getAuthClient, initInfo, initState } from '../../utils/init';
+import { authHookCb } from '../../integrations/stytch';
+import Client from '../../utils/client';
+import { isAuthStytch } from '../../integrations/utils';
+import Spinner from './Spinner';
+import { isHrefFirebaseMagicLink } from '../../integrations/firebase';
+/** TODO: These next 2 should maybe be dynamically imported, but having trouble with that
+ * combined 6.9k gzipped, so OK for now
+ */
+import { useIdleTimer } from 'react-idle-timer';
+import throttle from 'lodash.throttle';
+
+const TEN_SECONDS_IN_MILLISECONDS = 1000 * 10;
+const FIVE_MINUTES_IN_MILLISECONDS = 1000 * 60 * 5;
+
+export const authState = {
+  // This is a flag so we only redirect to the login start step immediately
+  // after auth, not during other form navigation
+  redirectAfterLogin: false,
+  sentAuth: false,
+  clearLoader: () => {},
+  onLogin: () => {},
+  onLogout: () => {}
+};
+
+const FeatheryAuthGate = ({
```

Comment:
is this the best prop name? other ideas include
* LoginFormWrapper
* LoginFormGate
* AuthFormGate
* AuthFormWrapper
* FormLogin
* FormAuth

could be a good idea to poll the team on some ideas to see what reads clearer for them